spring-kotlin/aws-ssh-server-manager/src/main/kotlin/dev/santos/awssshservermanager/config/WebSecurityConfig.kt

```diff
@@ -0,0 +1,54 @@
+package dev.santos.awssshservermanager.config
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.Customizer.withDefaults
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.crypto.password.PasswordEncoder
+
+
+@Configuration
+@EnableWebSecurity
+class WebSecurityConfig : WebSecurityConfigurerAdapter() {
+    @Autowired
+    private lateinit var userDetailsService: UserDetailsService
+
+    @Autowired
+    private lateinit var passwordEncoder: PasswordEncoder
+
+    @Throws(Exception::class)
+    override fun configure(http: HttpSecurity) {
+        http
+                .cors(withDefaults())
+                .csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/built/**",
+                        "/img/**",
+                        "/css/**",
+                        "/favicon.ico").permitAll()
+                .antMatchers("/api/v1/user/signup", "/api/v1/user/login").permitAll()
+                .antMatchers("/api/**").authenticated()
+                //.anyRequest().authenticated()
+                .and()
+                .formLogin()
+                .loginPage("/login.html")
+                .defaultSuccessUrl("/", true)
+                .permitAll()
+                .and()
+                .httpBasic()
+                .and()
+                .logout()
+                .logoutSuccessUrl("/")
+
+        http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+    }
+
+    override fun configure(auth: AuthenticationManagerBuilder) {
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder)
+    }
+}
```

Comment:
does kotlin have a linter? It's usual for editors to auto-add a new line at the end

Reply:
also the spacing in this file looks quite odd

Reply:
Good shout: it should have! Will try [this one](https://ktlint.github.io/)